fvtest/compilerunittest/p/BinaryEncoder.cpp

```diff
@@ -0,0 +1,2856 @@
+    std::make_tuple(TR::InstOpCode::bnun,  TR::InstOpCode::bad, BinaryInstruction()),
+    std::make_tuple(TR::InstOpCode::bun,   TR::InstOpCode::bad, BinaryInstruction())
+));
+
+// TODO: Disabled due to known issue with encoding mtfsfi. instructions
+/*INSTANTIATE_TEST_CASE_P(SprMove, PPCImm2EncodingTest, ::testing::Values(
+    std::make_tuple(TR::InstOpCode::mtfsfi, 0x0, 0x0, 0xfc01010cu),
+    std::make_tuple(TR::InstOpCode::mtfsfi, 0xf, 0x0, 0xfc01f10cu),
+    std::make_tuple(TR::InstOpCode::mtfsfi, 0x0, 0xf, 0xff80010cu),
+    std::make_tuple(TR::InstOpCode::mtfsfi, 0x0, 0x7, 0xff81010cu),
+    std::make_tuple(TR::InstOpCode::mtfsfi, 0x0, 0x8, 0xfc00010cu)
+));*/
+
+INSTANTIATE_TEST_CASE_P(SprMove, PPCTrg1EncodingTest, ::testing::Values(
+    std::make_tuple(TR::InstOpCode::mfcr,      TR::RealRegister::gr0,  0x7c000026u),
+    std::make_tuple(TR::InstOpCode::mfcr,      TR::RealRegister::gr31, 0x7fe00026u),
+    std::make_tuple(TR::InstOpCode::mfctr,     TR::RealRegister::gr0,  0x7c0902a6u),
+    std::make_tuple(TR::InstOpCode::mfctr,     TR::RealRegister::gr31, 0x7fe902a6u),
+    std::make_tuple(TR::InstOpCode::mffs,      TR::RealRegister::gr0,  0xfc00048eu),
+    std::make_tuple(TR::InstOpCode::mffs,      TR::RealRegister::gr31, 0xffe0048eu),
+    std::make_tuple(TR::InstOpCode::mflr,      TR::RealRegister::gr0,  0x7c0802a6u),
+    std::make_tuple(TR::InstOpCode::mflr,      TR::RealRegister::gr31, 0x7fe802a6u),
+    std::make_tuple(TR::InstOpCode::mftexasr,  TR::RealRegister::gr0,  0x7c0222a6u),
+    std::make_tuple(TR::InstOpCode::mftexasr,  TR::RealRegister::gr31, 0x7fe222a6u),
+    std::make_tuple(TR::InstOpCode::mftexasru, TR::RealRegister::gr0,  0x7c0322a6u),
+    std::make_tuple(TR::InstOpCode::mftexasru, TR::RealRegister::gr31, 0x7fe322a6u)
+));
+
+INSTANTIATE_TEST_CASE_P(SprMove, PPCTrg1ImmEncodingTest, ::testing::Values(
+    std::make_tuple(TR::InstOpCode::mcrfs, TR::RealRegister::cr7, 0, 0xff800080u),
+    std::make_tuple(TR::InstOpCode::mcrfs, TR::RealRegister::cr0, 7, 0xfc1c0080u)
+));
+
+INSTANTIATE_TEST_CASE_P(SprMove, PPCSrc1EncodingTest, ::testing::Values(
+    std::make_tuple(TR::InstOpCode::mtctr,  TR::RealRegister::gr0,     0u, 0x7c0903a6u),
+    std::make_tuple(TR::InstOpCode::mtctr,  TR::RealRegister::gr31,    0u, 0x7fe903a6u),
+    // TODO: Disabled due to known issue with encoding mtfsf., mtfsfl., and mtfsfw. instructions
+    /*std::make_tuple(TR::InstOpCode::mtfsf,  TR::RealRegister::fp0,  0xffu, 0xfdfe058eu),
+    std::make_tuple(TR::InstOpCode::mtfsf,  TR::RealRegister::fp31, 0x00u, 0xfc00fd8eu),
+    std::make_tuple(TR::InstOpCode::mtfsfl, TR::RealRegister::fp0,  0xffu, 0xfffe058eu),
+    std::make_tuple(TR::InstOpCode::mtfsfl, TR::RealRegister::fp31, 0x00u, 0xfe00fd8eu),
+    std::make_tuple(TR::InstOpCode::mtfsfw, TR::RealRegister::fp0,  0xffu, 0xfdff058eu),
+    std::make_tuple(TR::InstOpCode::mtfsfw, TR::RealRegister::fp31, 0x00u, 0xfc01fd8eu),*/
```

Comment:
We should open an issue to track this and reference it after the TODO:

```
// TODO (#9999): Disabled due to known issue with...
```

Reply:
While I'd usually agree, these issues are already implicitly fixed by the code for the upcoming Power binary encoder refactor (#4720, phase 3). The PR with those fixes is almost ready and is mostly just waiting on this and a couple other PRs to merge first, so I don't think opening an issue would really provide any value here, since it would be very short-lived.

Reply:
Ok, just to be clear, phase 3 will end up fixing and uncommenting these lines? If so then we can omit the TODO reference.

Reply:
Yes, that's correct. These lines are uncommented and verified to work after the phase 3 refactor.